erpnext_com/www/docs/user/manual/en/setting-up/articles/integrating-erpnext-with-biometric-attendance-devices.md

```diff
@@ -0,0 +1,49 @@
+ keywords: frappe, erpnext, erpnext attendance, biometric device integration, human resource, auto attendance
+---
+
+<!-- add-breadcrumbs -->
+# Integrating ERPNext With Biometric Attendance Devices
+
+The Attendance Punch Logs from the biometric device is a type of Check-in and Check-out log of an employee.
+
+ERPNext has a provision to store these logs in a document called Employee Checkin.
+
+Attendance can then be marked based on the Employee Checkin records and the shift [Auto Attendance Settings](/docs/user/manual/en/human-resources/shift-management#25-auto-attendance-settings) of the employee by using [Auto Attendance](/docs/user/manual/en/human-resources/auto-attendance)
+
+Hence, integrating your Biometric Device --or any access control system that collects IN/OUT--, can be done by the following steps:
+
+  1. [Setting up Auto Attendance to mark attendance from the Employee Checkin](#1-setting-up-auto-attendance-to-mark-attendance-from-the-employee-checkin)
+  1. [Populating the Biometric Punch Logs into ERPNext's Employee Checkin](#2-populating-the-biometric-punch-logs-into-erpnexts-employee-checkin)
+
+## 1. Setting up Auto Attendance to mark attendance from the Employee Checkin
+
+Before you import employee's Check-in and Check-out logs into your ERPNext system. You would have to set up the Employees and their shifts to be able to generate attendance using the Auto Attendance feature in ERPNext.
+
+Please refer to the following link to setup Auto Attendance: [Setup Auto Attendance](/docs/user/manual/en/human-resources/auto-attendance#steps-to-setup-auto-attendance)
+
+Once you have set up the Employee and assigned shifts to employees, you are now ready to proceed to the next step.
+
+## 2. Populating the Biometric Punch Logs into ERPNext's Employee Checkin
+Depending on your biometric system and its features, there can be a lot of ways you can import the Punch logs into ERPNext:
+
+1. Use ERPNext's Data Import Tool:
+    - The simplest possible solution(in terms of implementation complexity) would be to generate an Excel/CSV of the Check-in/Check-out and use the built-in data import tool in ERPNext to periodically import to your Employee Checkin Document
+    - Please refer to ERPNext's Documentation on Data Import Tool for more on how to do this.
+
+1. API Integration:
+    - You can automate the process of importing the Biometric Punch Logs but integrating with the available API in ERPNext.
+    - This API can be accessed at: `/api/method/erpnext.hr.doctype.employee_checkin.employee_checkin.add_log_based_on_employee_field`
+    - This method requires technical knowledge and you should probably get in touch with your ERPNext implementor or Biometric system vendor.
+
+1. Setup a python script on your computer to integrate ZKTeco or similar devices:
```

Comment:
Set up -verb
Setup - noun
Here it should be set up

Reply:
I see... done.